privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardHybridViewModel.kt

```diff
@@ -0,0 +1,215 @@
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.BufferOverflow.DROP_OLDEST
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import java.util.*
+import javax.inject.Inject
+
+@ContributesViewModel(ActivityScope::class)
+class PrivacyDashboardHybridViewModel @Inject constructor(
+    private val userWhitelistDao: UserWhitelistDao,
+    private val contentBlocking: ContentBlocking,
+    private val pixel: Pixel,
+    private val dispatcher: DispatcherProvider,
+    @AppCoroutineScope private val appCoroutineScope: CoroutineScope,
+    private val siteProtectionsViewStateMapper: SiteProtectionsViewStateMapper
+) : ViewModel() {
+
+    private val command = Channel<Command>(1, DROP_OLDEST)
+
+    sealed class Command {
+        class LaunchReportBrokenSite(val data: BrokenSiteData) : Command()
+    }
+
+    data class ViewState(
+        val userSettingsViewState: UserSettingsViewState,
+        val siteProtectionsViewState: SiteProtectionsViewState,
+        val userChangedValues: Boolean = false
+    )
+
+    data class UserSettingsViewState(
+        val privacyProtectionEnabled: Boolean
+    )
+
+    data class SiteProtectionsViewState(
+        val url: String,
+        val status: String = "complete",
+        val upgradedHttps: Boolean,
+        val parentEntity: EntityViewState?,
+        val site: SiteViewState,
+        val trackers: Map<String, TrackerViewState>,
+        val trackerBlocked: Map<String, TrackerViewState>,
+        val secCertificateViewModels: List<CertificateViewState?> = emptyList(),
+        val locale: String = Locale.getDefault().language
+    )
+
+    data class CertificateViewState(
+        val commonName: String,
+        val publicKey: PublicKeyViewState? = null,
+        val emails: List<String> = emptyList(),
+        val summary: String? = null
+    )
+
+    data class PublicKeyViewState(
+        val blockSize: Int?,
+        val canEncrypt: Boolean?,
+        val bitSize: Int?,
+        val canSign: Boolean?,
+        val canDerive: Boolean?,
+        val canUnwrap: Boolean?,
+        val canWrap: Boolean?,
+        val canDecrypt: Boolean?,
+        val effectiveSize: Int?,
+        val isPermanent: Boolean?,
+        val type: String?,
+        val externalRepresentation: String?,
+        val canVerify: Boolean?,
+        val keyId: String?
+    )
+
+    data class EntityViewState(
+        val displayName: String,
+        val prevalence: Double
+    )
+
+    data class SiteViewState(
+        val url: String,
+        val domain: String,
+        val trackersUrls: Set<String>,
+        val whitelisted: Boolean,
```

Comment:
We should not be using terms like whitelisted, suggest to change to allowlisted.

Reply:
This is now changed. 👍